[Automation] javac compile fails for com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0

## javac compile on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26742420600
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8
Raw versions JSON: [
  "2.22.0"
]
Parsed version count: 1
Parsed versions: 2.22.0
Timeout per Gradle invocation: 5m
========================================================================================
 Testing com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8:2.22.0
========================================================================================
========================================================================================
 com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8:2.22.0 stage native-image run
========================================================================================
Starting stage 'native-image run' for 2.22.0 at 2026-06-01T08:04:19Z
Command: env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.9.8/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.9.8/build/reports/problems/problems-report.html
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':nativeTestCompile'.
> Could not resolve all dependencies for configuration ':nativeImageTestClasspath'.
   > Could not resolve all dependencies for configuration ':nativeImageTestClasspath'.
      > Could not find com.fasterxml.jackson.core:jackson-core:2.22.0.
        Required by:
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson:jackson-bom:2.22.0
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:2.22.0
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson.module:jackson-module-jaxb-annotations:2.22.0
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:2.22.0 > com.fasterxml.jackson.core:jackson-databind:2.22.0

* Try:
> The project declares repositories, effectively ignoring the repositories you have declared in the settings.
   To determine how project repositories are declared, configure your build to fail on project repositories.
   For more information, please refer to https://docs.gradle.org/9.1.0/userguide/declaring_repositories.html#sub:fail_build_on_project_repositories in the Gradle documentation.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 2s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
8 actionable tasks: 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 11s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.22.0 at 2026-06-01T08:04:31Z with exit code 1 after 12s
❌ ERROR: 'native-image run' for 2.22.0 failed with exit code 1.
SOFT-FAIL[native-image run][2.22.0][env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8]
========================================================================================
 com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8:2.22.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.22.0 at 2026-06-01T08:04:31Z
Command: env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.9.8/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 892ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.9.8/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':compileTestJava'.
> Could not resolve all dependencies for configuration ':testCompileClasspath'.
   > Could not resolve all dependencies for configuration ':testCompileClasspath'.
      > Could not find com.fasterxml.jackson.core:jackson-core:2.22.0.
        Searched in the following locations:
          - file:/home/runner/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.22.0/jackson-core-2.22.0.pom
          - https://repo.maven.apache.org/maven2/com/fasterxml/jackson/core/jackson-core/2.22.0/jackson-core-2.22.0.pom
        Required by:
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson:jackson-bom:2.22.0
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:2.22.0
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson.module:jackson-module-jaxb-annotations:2.22.0
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:2.22.0 > com.fasterxml.jackson.core:jackson-databind:2.22.0

* Try:
> The project declares repositories, effectively ignoring the repositories you have declared in the settings.
   To determine how project repositories are declared, configure your build to fail on project repositories.
   For more information, please refer to https://docs.gradle.org/9.1.0/userguide/declaring_repositories.html#sub:fail_build_on_project_repositories in the Gradle documentation.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
8 actionable tasks: 8 up-to-date

BUILD FAILED in 945ms

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.22.0 at 2026-06-01T08:04:34Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 2.22.0 failed with exit code 1.
FAILED[javac compile][2.22.0][env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26742420600
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8
Raw versions JSON: [
  "2.22.0"
]
Parsed version count: 1
Parsed versions: 2.22.0
Timeout per Gradle invocation: 5m
========================================================================================
 Testing com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8:2.22.0
========================================================================================
========================================================================================
 com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8:2.22.0 stage native-image run
========================================================================================
Starting stage 'native-image run' for 2.22.0 at 2026-06-01T08:04:18Z
Command: env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.9.8/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.9.8/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':nativeTestCompile'.
> Could not resolve all dependencies for configuration ':nativeImageTestClasspath'.
   > Could not resolve all dependencies for configuration ':nativeImageTestClasspath'.
      > Could not find com.fasterxml.jackson.core:jackson-core:2.22.0.
        Required by:
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson:jackson-bom:2.22.0
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:2.22.0
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson.module:jackson-module-jaxb-annotations:2.22.0
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:2.22.0 > com.fasterxml.jackson.core:jackson-databind:2.22.0

* Try:
> The project declares repositories, effectively ignoring the repositories you have declared in the settings.
   To determine how project repositories are declared, configure your build to fail on project repositories.
   For more information, please refer to https://docs.gradle.org/9.1.0/userguide/declaring_repositories.html#sub:fail_build_on_project_repositories in the Gradle documentation.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.


BUILD FAILED in 3s
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
8 actionable tasks: 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 11s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.22.0 at 2026-06-01T08:04:29Z with exit code 1 after 11s
❌ ERROR: 'native-image run' for 2.22.0 failed with exit code 1.
SOFT-FAIL[native-image run][2.22.0][env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8]
========================================================================================
 com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8:2.22.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.22.0 at 2026-06-01T08:04:29Z
Command: env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.9.8/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 902ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.9.8/build/reports/problems/problems-report.html
FAILURE: Build failed with an exception.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

* What went wrong:

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.Could not determine the dependencies of task ':compileTestJava'.
> Could not resolve all dependencies for configuration ':testCompileClasspath'.
   > Could not resolve all dependencies for configuration ':testCompileClasspath'.
      > Could not find com.fasterxml.jackson.core:jackson-core:2.22.0.
        Searched in the following locations:
          - file:/home/runner/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.22.0/jackson-core-2.22.0.pom
          - https://repo.maven.apache.org/maven2/com/fasterxml/jackson/core/jackson-core/2.22.0/jackson-core-2.22.0.pom
        Required by:
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson:jackson-bom:2.22.0

8 actionable tasks: 8 up-to-date
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:2.22.0
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson.module:jackson-module-jaxb-annotations:2.22.0
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:2.22.0 > com.fasterxml.jackson.core:jackson-databind:2.22.0

* Try:
> The project declares repositories, effectively ignoring the repositories you have declared in the settings.
   To determine how project repositories are declared, configure your build to fail on project repositories.
   For more information, please refer to https://docs.gradle.org/9.1.0/userguide/declaring_repositories.html#sub:fail_build_on_project_repositories in the Gradle documentation.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1s

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.22.0 at 2026-06-01T08:04:33Z with exit code 1 after 4s
❌ ERROR: 'javac compile' for 2.22.0 failed with exit code 1.
FAILED[javac compile][2.22.0][env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26742420600
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8
Raw versions JSON: [
  "2.22.0"
]
Parsed version count: 1
Parsed versions: 2.22.0
Timeout per Gradle invocation: 5m
========================================================================================
 Testing com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8:2.22.0
========================================================================================
========================================================================================
 com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8:2.22.0 stage native-image run
========================================================================================
Starting stage 'native-image run' for 2.22.0 at 2026-06-01T08:04:17Z
Command: env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.9.8/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.9.8/build/reports/problems/problems-report.html


FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':nativeTestCompile'.
> Could not resolve all dependencies for configuration ':nativeImageTestClasspath'.
   > Could not resolve all dependencies for configuration ':nativeImageTestClasspath'.
      > Could not find com.fasterxml.jackson.core:jackson-core:2.22.0.
        Required by:
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson:jackson-bom:2.22.0
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:2.22.0
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson.module:jackson-module-jaxb-annotations:2.22.0
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:2.22.0 > com.fasterxml.jackson.core:jackson-databind:2.22.0

* Try:
> The project declares repositories, effectively ignoring the repositories you have declared in the settings.
   To determine how project repositories are declared, configure your build to fail on project repositories.
   For more information, please refer to https://docs.gradle.org/9.1.0/userguide/declaring_repositories.html#sub:fail_build_on_project_repositories in the Gradle documentation.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
BUILD FAILED
 in 2s
8 actionable tasks: 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 11s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.22.0 at 2026-06-01T08:04:29Z with exit code 1 after 12s
❌ ERROR: 'native-image run' for 2.22.0 failed with exit code 1.
SOFT-FAIL[native-image run][2.22.0][env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8]
========================================================================================
 com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8:2.22.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.22.0 at 2026-06-01T08:04:29Z
Command: env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.9.8/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 879ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.9.8/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':compileTestJava'.
> Could not resolve all dependencies for configuration ':testCompileClasspath'.
   > Could not resolve all dependencies for configuration ':testCompileClasspath'.
      > Could not find com.fasterxml.jackson.core:jackson-core:2.22.0.
        Searched in the following locations:
          - file:/home/runner/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.22.0/jackson-core-2.22.0.pom
          - https://repo.maven.apache.org/maven2/com/fasterxml/jackson/core/jackson-core/2.22.0/jackson-core-2.22.0.pom
        Required by:
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson:jackson-bom:2.22.0
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:2.22.0
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson.module:jackson-module-jaxb-annotations:2.22.0
            root project 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests' > com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0 > com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:2.22.0 > com.fasterxml.jackson.core:jackson-databind:2.22.0

* Try:
> The project declares repositories, effectively ignoring the repositories you have declared in the settings.
   To determine how project repositories are declared, configure your build to fail on project repositories.
   For more information, please refer to https://docs.gradle.org/9.1.0/userguide/declaring_repositories.html#sub:fail_build_on_project_repositories in the Gradle documentation.
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 935ms

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
8 actionable tasks: 8 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.22.0 at 2026-06-01T08:04:32Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 2.22.0 failed with exit code 1.
FAILED[javac compile][2.22.0][env GVM_TCK_LV=2.22.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8]
```

--- a/metadata/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/reachability-metadata.json
+++ b/metadata/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/reachability-metadata.json
@@ -1,0 +1,119 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      },
+      "type": "com.fasterxml.jackson.databind.ext.Java7SupportImpl",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.fasterxml.jackson.jaxrs.json.JsonMapperConfigurator"
+      },
+      "type": "com.fasterxml.jackson.module.jaxb.JaxbAnnotationIntrospector",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.fasterxml.jackson.jaxrs.json.JsonMapperConfigurator"
+      },
+      "type": "com.fasterxml.jackson.module.jaxb.deser.DataHandlerJsonDeserializer",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com.fasterxml.jackson.jaxrs.json.JsonMapperConfigurator"
+      },
+      "type": "com.fasterxml.jackson.module.jaxb.ser.DataHandlerJsonSerializer",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      },
+      "type": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest$JaxbPayload",
+      "fields": [
+        {
+          "name": "message"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      },
+      "type": "java.io.Serializable"
+    },
+    {
+      "condition": {
+        "typeReached": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      },
+      "type": "java.lang.Class",
+      "methods": [
+        {
+          "name": "isRecord",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      },
+      "type": "java.util.AbstractMap"
+    },
+    {
+      "condition": {
+        "typeReached": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      },
+      "type": "java.util.ImmutableCollections$AbstractImmutableMap"
+    },
+    {
+      "condition": {
+        "typeReached": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      },
+      "type": "java.util.ImmutableCollections$Map1"
+    },
+    {
+      "condition": {
+        "typeReached": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      },
+      "type": "java.util.Map"
+    }
+  ],
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      },
+      "glob": "META-INF/services/org.assertj.core.configuration.Configuration"
+    },
+    {
+      "condition": {
+        "typeReached": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      },
+      "glob": "META-INF/services/org.assertj.core.presentation.Representation"
+    }
+  ]
+}

--- a/metadata/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/reachability-metadata.json
+++ b/metadata/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/reachability-metadata.json
@@ -1,119 +1,40 @@
 {
-  "reflection": [
+  "reflection" : [
     {
-      "condition": {
-        "typeReached": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      "condition" : {
+        "typeReached" : "com.fasterxml.jackson.jaxrs.json.JsonMapperConfigurator"
       },
-      "type": "com.fasterxml.jackson.databind.ext.Java7SupportImpl",
-      "methods": [
+      "type" : "com.fasterxml.jackson.module.jaxb.JaxbAnnotationIntrospector",
+      "methods" : [
         {
-          "name": "<init>",
-          "parameterTypes": []
+          "name" : "<init>",
+          "parameterTypes" : [ ]
         }
       ]
     },
     {
-      "condition": {
-        "typeReached": "com.fasterxml.jackson.jaxrs.json.JsonMapperConfigurator"
+      "condition" : {
+        "typeReached" : "com.fasterxml.jackson.jaxrs.json.JsonMapperConfigurator"
       },
-      "type": "com.fasterxml.jackson.module.jaxb.JaxbAnnotationIntrospector",
-      "methods": [
+      "type" : "com.fasterxml.jackson.module.jaxb.deser.DataHandlerJsonDeserializer",
+      "methods" : [
         {
-          "name": "<init>",
-          "parameterTypes": []
+          "name" : "<init>",
+          "parameterTypes" : [ ]
         }
       ]
     },
     {
-      "condition": {
-        "typeReached": "com.fasterxml.jackson.jaxrs.json.JsonMapperConfigurator"
+      "condition" : {
+        "typeReached" : "com.fasterxml.jackson.jaxrs.json.JsonMapperConfigurator"
       },
-      "type": "com.fasterxml.jackson.module.jaxb.deser.DataHandlerJsonDeserializer",
-      "methods": [
+      "type" : "com.fasterxml.jackson.module.jaxb.ser.DataHandlerJsonSerializer",
+      "methods" : [
         {
-          "name": "<init>",
-          "parameterTypes": []
+          "name" : "<init>",
+          "parameterTypes" : [ ]
         }
       ]
-    },
-    {
-      "condition": {
-        "typeReached": "com.fasterxml.jackson.jaxrs.json.JsonMapperConfigurator"
-      },
-      "type": "com.fasterxml.jackson.module.jaxb.ser.DataHandlerJsonSerializer",
-      "methods": [
-        {
-          "name": "<init>",
-          "parameterTypes": []
-        }
-      ]
-    },
-    {
-      "condition": {
-        "typeReached": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
-      },
-      "type": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest$JaxbPayload",
-      "fields": [
-        {
-          "name": "message"
-        }
-      ]
-    },
-    {
-      "condition": {
-        "typeReached": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
-      },
-      "type": "java.io.Serializable"
-    },
-    {
-      "condition": {
-        "typeReached": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
-      },
-      "type": "java.lang.Class",
-      "methods": [
-        {
-          "name": "isRecord",
-          "parameterTypes": []
-        }
-      ]
-    },
-    {
-      "condition": {
-        "typeReached": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
-      },
-      "type": "java.util.AbstractMap"
-    },
-    {
-      "condition": {
-        "typeReached": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
-      },
-      "type": "java.util.ImmutableCollections$AbstractImmutableMap"
-    },
-    {
-      "condition": {
-        "typeReached": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
-      },
-      "type": "java.util.ImmutableCollections$Map1"
-    },
-    {
-      "condition": {
-        "typeReached": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
-      },
-      "type": "java.util.Map"
-    }
-  ],
-  "resources": [
-    {
-      "condition": {
-        "typeReached": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
-      },
-      "glob": "META-INF/services/org.assertj.core.configuration.Configuration"
-    },
-    {
-      "condition": {
-        "typeReached": "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
-      },
-      "glob": "META-INF/services/org.assertj.core.presentation.Representation"
     }
   ]
 }

--- a/metadata/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/index.json
+++ b/metadata/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/index.json
@@ -1,6 +1,15 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "2.22.0",
+    "tested-versions" : [
+      "2.22.0"
+    ],
+    "allowed-packages" : [
+      "com.fasterxml.jackson.jaxrs.json"
+    ]
+  },
+  {
     "metadata-version" : "2.9.8",
     "source-code-url" : "https://repo.maven.apache.org/maven2/com/fasterxml/jackson/jaxrs/jackson-jaxrs-json-provider/$version$/jackson-jaxrs-json-provider-$version$-sources.jar",
     "repository-url" : "https://github.com/FasterXML/jackson-jaxrs-providers",

--- a/metadata/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/index.json
+++ b/metadata/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "2.22.0",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/com/fasterxml/jackson/jaxrs/jackson-jaxrs-json-provider/$version$/jackson-jaxrs-json-provider-$version$-sources.jar",
+    "repository-url" : "https://github.com/FasterXML/jackson-jaxrs-providers",
+    "test-code-url" : "https://github.com/FasterXML/jackson-jaxrs-providers/tree/jackson-jaxrs-providers-$version$/json/src/test",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/com/fasterxml/jackson/jaxrs/jackson-jaxrs-json-provider/$version$/jackson-jaxrs-json-provider-$version$-javadoc.jar",
+    "description" : "Jackson JAX-RS JSON Provider integrates Jackson databinding with JAX-RS implementations such as Jersey and RESTEasy to read and write JSON request and response bodies. It supplies MessageBodyReader and MessageBodyWriter providers plus related JSON/JAXB configuration support for REST endpoints.",
     "tested-versions" : [
       "2.22.0"
     ],

--- a/stats/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/execution-metrics.json
+++ b/stats/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/execution-metrics.json
@@ -1,0 +1,100 @@
+{
+  "fix_javac_fail:2026-06-05": {
+    "timestamp": "2026-06-05T06:08:15.784913Z",
+    "library": "com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0",
+    "starting_commit": "8f714a9b0a0734f73a610dc9e1adc72e43991bdb",
+    "ending_commit": "6f453e01c3eabfbc6b6a36e2c3b03ae097d625a2",
+    "previous_library": "com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.9.8",
+    "strategy_name": "javac_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "2.22.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 1,
+        "totalCalls": 1
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 104,
+          "missed": 333,
+          "ratio": 0.237986,
+          "total": 437
+        },
+        "line": {
+          "covered": 24,
+          "missed": 84,
+          "ratio": 0.222222,
+          "total": 108
+        },
+        "method": {
+          "covered": 4,
+          "missed": 29,
+          "ratio": 0.121212,
+          "total": 33
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "2.9.8",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 1,
+        "totalCalls": 1
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 86,
+          "missed": 372,
+          "ratio": 0.187773,
+          "total": 458
+        },
+        "line": {
+          "covered": 19,
+          "missed": 94,
+          "ratio": 0.168142,
+          "total": 113
+        },
+        "method": {
+          "covered": 4,
+          "missed": 30,
+          "ratio": 0.117647,
+          "total": 34
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 85163,
+      "cached_input_tokens_used": 839168,
+      "output_tokens_used": 7974,
+      "iterations": 2,
+      "cost_usd": 0.6588,
+      "tested_library_loc": 24,
+      "metadata_entries": 6,
+      "test_only_metadata_entries": 13,
+      "previous_library_metadata_entries": 10,
+      "code_coverage_percent": 22.22,
+      "previous_library_coverage_percent": 16.81
+    },
+    "artifacts": {
+      "test_file": "tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/src/test/java/com_fasterxml_jackson_jaxrs/jackson_jaxrs_json_provider/JsonMapperConfiguratorTest.java",
+      "metadata_file": "metadata/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/stats.json
+++ b/stats/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/stats.json
@@ -1,0 +1,37 @@
+{
+  "versions" : [ {
+    "version" : "2.22.0",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 1,
+          "totalCalls" : 1
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 1,
+      "totalCalls" : 1
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 104,
+        "missed" : 333,
+        "ratio" : 0.237986,
+        "total" : 437
+      },
+      "line" : {
+        "covered" : 24,
+        "missed" : 84,
+        "ratio" : 0.222222,
+        "total" : 108
+      },
+      "method" : {
+        "covered" : 4,
+        "missed" : 29,
+        "ratio" : 0.121212,
+        "total" : 33
+      }
+    }
+  } ]
+}

--- a/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/.gitignore
+++ b/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/build.gradle
+++ b/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/build.gradle
@@ -1,0 +1,28 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:$libraryVersion"
+    testImplementation 'javax.xml.bind:jaxb-api:2.3.1'
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/gradle.properties
+++ b/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.22.0
+library.version = 2.22.0
+metadata.dir = com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/

--- a/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/settings.gradle
+++ b/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider_tests'

--- a/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/src/test/java/com_fasterxml_jackson_jaxrs/jackson_jaxrs_json_provider/JsonMapperConfiguratorTest.java
+++ b/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/src/test/java/com_fasterxml_jackson_jaxrs/jackson_jaxrs_json_provider/JsonMapperConfiguratorTest.java
@@ -8,16 +8,36 @@ package com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider;
 
 import com.fasterxml.jackson.databind.AnnotationIntrospector;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.introspect.JacksonAnnotationIntrospector;
 import com.fasterxml.jackson.jaxrs.cfg.Annotations;
 import com.fasterxml.jackson.jaxrs.json.JsonMapperConfigurator;
 import com.fasterxml.jackson.module.jaxb.JaxbAnnotationIntrospector;
 import org.junit.jupiter.api.Test;
 
+import java.util.Map;
+
+import javax.xml.bind.annotation.XmlElement;
+
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class JsonMapperConfiguratorTest {
     @Test
-    public void getDefaultMapperInstantiatesJaxbAnnotationIntrospector() {
+    public void getDefaultMapperConfiguresJacksonAnnotationIntrospector() throws Exception {
+        JsonMapperConfigurator configurator = new JsonMapperConfigurator(
+                null,
+                new Annotations[] {Annotations.JACKSON});
+
+        ObjectMapper mapper = configurator.getDefaultMapper();
+
+        AnnotationIntrospector introspector = mapper.getSerializationConfig()
+                .getAnnotationIntrospector();
+        assertThat(introspector).isInstanceOf(JacksonAnnotationIntrospector.class);
+        assertThat(mapper.writeValueAsString(Map.of("message", "hello"))).isEqualTo("{\"message\":\"hello\"}");
+        assertThat(configurator.getDefaultMapper()).isSameAs(mapper);
+    }
+
+    @Test
+    public void getDefaultMapperConfiguresJaxbAnnotationIntrospector() throws Exception {
         JsonMapperConfigurator configurator = new JsonMapperConfigurator(
                 null,
                 new Annotations[] {Annotations.JAXB});
@@ -27,6 +47,17 @@ public class JsonMapperConfiguratorTest {
         AnnotationIntrospector introspector = mapper.getSerializationConfig()
                 .getAnnotationIntrospector();
         assertThat(introspector).isInstanceOf(JaxbAnnotationIntrospector.class);
+        assertThat(mapper.writeValueAsString(new JaxbPayload("hello")))
+                .isEqualTo("{\"jaxbMessage\":\"hello\"}");
         assertThat(configurator.getDefaultMapper()).isSameAs(mapper);
+    }
+
+    public static class JaxbPayload {
+        @XmlElement(name = "jaxbMessage")
+        public String message;
+
+        public JaxbPayload(String message) {
+            this.message = message;
+        }
     }
 }

--- a/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/src/test/java/com_fasterxml_jackson_jaxrs/jackson_jaxrs_json_provider/JsonMapperConfiguratorTest.java
+++ b/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/src/test/java/com_fasterxml_jackson_jaxrs/jackson_jaxrs_json_provider/JsonMapperConfiguratorTest.java
@@ -1,0 +1,32 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider;
+
+import com.fasterxml.jackson.databind.AnnotationIntrospector;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.jaxrs.cfg.Annotations;
+import com.fasterxml.jackson.jaxrs.json.JsonMapperConfigurator;
+import com.fasterxml.jackson.module.jaxb.JaxbAnnotationIntrospector;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class JsonMapperConfiguratorTest {
+    @Test
+    public void getDefaultMapperInstantiatesJaxbAnnotationIntrospector() {
+        JsonMapperConfigurator configurator = new JsonMapperConfigurator(
+                null,
+                new Annotations[] {Annotations.JAXB});
+
+        ObjectMapper mapper = configurator.getDefaultMapper();
+
+        AnnotationIntrospector introspector = mapper.getSerializationConfig()
+                .getAnnotationIntrospector();
+        assertThat(introspector).isInstanceOf(JaxbAnnotationIntrospector.class);
+        assertThat(configurator.getDefaultMapper()).isSameAs(mapper);
+    }
+}

--- a/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -1,0 +1,83 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      },
+      "type" : "com.fasterxml.jackson.databind.ext.Java7SupportImpl",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      },
+      "type" : "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest$JaxbPayload",
+      "fields" : [
+        {
+          "name" : "message"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      },
+      "type" : "java.io.Serializable"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      },
+      "type" : "java.lang.Class",
+      "methods" : [
+        {
+          "name" : "isRecord",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      },
+      "type" : "java.util.AbstractMap"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      },
+      "type" : "java.util.ImmutableCollections$AbstractImmutableMap"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      },
+      "type" : "java.util.ImmutableCollections$Map1"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      },
+      "type" : "java.util.Map"
+    }
+  ],
+  "resources" : [
+    {
+      "condition" : {
+        "typeReached" : "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      },
+      "glob" : "META-INF/services/org.assertj.core.configuration.Configuration"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.JsonMapperConfiguratorTest"
+      },
+      "glob" : "META-INF/services/org.assertj.core.presentation.Representation"
+    }
+  ]
+}

--- a/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/user-code-filter.json
+++ b/tests/src/com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/2.22.0/user-code-filter.json
@@ -1,0 +1,13 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "com.fasterxml.jackson.jaxrs.json.**"
+    },
+    {
+      "includeClasses" : "com_fasterxml_jackson_jaxrs.jackson_jaxrs_json_provider.**"
+    }
+  ]
+}

